src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -176,11 +229,25 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
                         assembly, filter, namespaceFilter, internalsVisible,
                         counter, cancellationToken);
 
-                    GetExtensionMethodItemsWorker(position, semanticModel, receiverTypeSymbol, matchingMethodSymbols, isSymbolFromCurrentCompilation: false, builder, namespaceNameCache);
+                    GetExtensionMethodItemsWorker(
+                        position, semanticModel, receiverTypeSymbol, matchingMethodSymbols,
+                        isSymbolFromCurrentCompilation: true, completionItemsbuilder, namespaceNameCache, cancellationToken);
                 }
             }
 
-            return builder.ToImmutable();
+            return completionItemsbuilder.ToImmutable();
+
+            // Add string represent complex types (i.e. "" and "[]") to the receiver type, so we would include in the filter
+            // info about extension methods with complex receiver type.
+            static ImmutableArray<string> AttachComplexTypes(ImmutableArray<string> receiverTypeNames)
+            {
+                using var _ = ArrayBuilder<string>.GetInstance(receiverTypeNames.Length + 2, out var receiverTypeNamesBuilder);
+                receiverTypeNamesBuilder.AddRange(receiverTypeNames);
+                receiverTypeNamesBuilder.Add(string.Empty);
+                receiverTypeNamesBuilder.Add("[]");
```

Comment:
need constants here.  one way to think about this is: if you decided to change this in one place, woudl you have to update many others?  if so, there should be central location for these.